infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -497,6 +502,8 @@ private void stopReplication(Set<LogReplicationSession> sessions) {
                 incomingMsgHandler.stopSinkManagerForSession(session);
             }
             this.sessions.remove(session);
+            this.configManager.getSessionToConfigMap().remove(session);
+            this.configManager.getSessionLockMap().remove(session);
             if (incomingSessions.contains(session)) {
                 incomingMsgHandler.stopSinkManagerForSession(session);
```

Comment:
Do you know why it is called twice in this method?

incomingMsgHandler.stopSinkManagerForSession(session) ?